crates/cli/src/main.rs

```diff
@@ -136,31 +136,53 @@ fn main() -> io::Result<()> {
             Ok(0)
         }
         Some((CMD_PREPROCESS_HOST, matches)) => {
-            let input_path = matches.get_one::<PathBuf>(ROC_FILE).unwrap();
+            let preprocess_host_err =
+                { |msg: String| user_error!("\n\n ERROR PRE-PROCESSING HOST: {}\n\n", msg) };
+
+            let host_path = matches.get_one::<PathBuf>(FLAG_PP_HOST).unwrap();
+            if !host_path.is_file() {
+                preprocess_host_err(format!(
+                    "Expected to find the host executable file at {}",
+                    &host_path.display()
+                ));
+            }
+
+            let platform_path = matches.get_one::<PathBuf>(FLAG_PP_PLATFORM).unwrap();
+            if !platform_path.is_file() {
+                preprocess_host_err(format!(
+                    "Expected to find the platform/main.roc file at {}",
+                    &platform_path.display()
+                ));
+            }
+
+            let dylib_path = matches.get_one::<PathBuf>(FLAG_PP_DYLIB).unwrap();
+            if !dylib_path.is_file() {
+                preprocess_host_err(format!(
+                    "Expected to find the app stub dynamic library file at {}",
+                    dylib_path.display()
+                ));
+            }
             let target = matches
                 .get_one::<String>(FLAG_TARGET)
                 .and_then(|s| Target::from_str(s).ok())
                 .unwrap_or_default();
 
-            let function_kind = FunctionKind::LambdaSet;
-            let (platform_path, stub_lib, stub_dll_symbols) = roc_linker::generate_stub_lib(
-                input_path,
-                RocCacheDir::Persistent(cache::roc_cache_dir().as_path()),
-                target,
-                function_kind,
-            );
+            let verbose_and_time = matches.get_one::<bool>(roc_cli::FLAG_VERBOSE).unwrap();
+
+            #[cfg(target_os = "windows")]
+            {
+                internal_error!("TODO populate stub_dll_symbols for Windows");
+            }
 
-            // TODO: pipeline the executable location through here.
-            // Currently it is essentally hardcoded as platform_path/dynhost.
             roc_linker::preprocess_host(
                 target,
-                &platform_path.with_file_name("main.roc"),
-                // The target triple string must be derived from the triple to convert from the generic
-                // `system` target to the exact specific target.
-                &platform_path.with_file_name(format!("{}.rh", target)),
-                &stub_lib,
-                &stub_dll_symbols,
```

Comment:
Here we are no longer passing the `stub_dll_symbols` down. But these appear only to be used in `crate::pe::preprocess_windows`. I'm not sure how to proceed.